src/core/node.ts

```diff
@@ -103,7 +112,7 @@ export abstract class Node /* TODO: implements INode*/ {
     // TODO: needs improvements, now called too often. Should be set propertly when applying snapshots / calling factories immediately!
     // TODO: should not be possible to change just subpath?
     setParent(newParent: Node | null, subpath: string | null = null) {
-        if (this.parent === newParent && this.pathParts[this.pathParts.length - 1] === subpath)
```

Comment:
Not sure whether this can be removed; item might be moved within same parent? Or maybe this is always the combination of two operations indeed